types/async/index.d.ts

```diff
@@ -259,7 +259,7 @@ export function race<T, E = Error>(tasks: Array<AsyncFunction<T, E>>, callback:
 export function memoize(fn: Function, hasher?: Function): Function;
 export function unmemoize(fn: Function): Function;
 export function ensureAsync(fn: (... argsAndCallback: any[]) => void): Function;
-export function constant(...values: any[]): Function;
+export function constant(...values: any[]): AsyncFunction<any>;
```

Comment:
Actually, it looks like
```ts
export function constant<T>(...values: T[]): AsyncFunctionEx<T>;
```
would be an improvement?

Reply:
Ahh, the `Ex` suffix here put me on the right track to notice that `AsyncFunction` was being defined locally, rather than being the canonical `AsyncFunction` type:

```
> async () => {}
[AsyncFunction]
```

Reply:
@andrewbranch thanks for the observation. In fact, `constant()` can receive arguments of any type. I don't think assuming those will be all of the same type is correct.

Reply:
In that case, the well-typed version would be

```ts
interface AsyncFunctionVariadic<T extends any[], E = Error> {
  (callback: (err?: E, ...params: T) => void): void;
}

export function constant<T extends any[]>(...values: T): AsyncFunctionVariadic<T>;
```

However, it would require bumping the minimum TypeScript version to 3.0 (currently it’s set to 2.3). I’m not an `async` user so I don’t have the context to know whether that tradeoff is worth it. It seems that this library could really benefit from these features (e.g. `compose`, `seq`, `applyEach` all suffer from `any`-infected typings that could be fixed with generic tuples like this). But of course, this is outside the scope of what you originally came here to do.

If you’re happy with the status quo here, I’ll go ahead and merge.

Reply:
@andrewbranch I think we should go ahead and merge.

Any further optimization like the one you mentioned would be great but should be addressed in conjunction with the `async` authors/maintainers IMO.

Reply:
I was wondering if you would be inspired to become a maintainer, since none of the maintainers showed up to review your PR 😄